autogpt/llm/modelsinfo.py

```diff
@@ -1,6 +1,8 @@
 COSTS = {
     "gpt-3.5-turbo": {"prompt": 0.002, "completion": 0.002},
     "gpt-3.5-turbo-0301": {"prompt": 0.002, "completion": 0.002},
+    "gpt-35-turbo": {"prompt": 0.002, "completion": 0.002},
+    "gpt-35-turbo-0301": {"prompt": 0.002, "completion": 0.002},
```

Comment:
Yes, the two new entries to `modelsinfo.py` is for the Azure API model name,  Azure specifies the model(gtp-35-turbo-0301) as gpt-35-turbo,  the second new entry is added to reserved for Azure may be changed to gpt-35-turbo-0301 in future, maybe I should delete it now.

For the change in `llm.utils.py`, because the model was initialized to none，when the model used by update_cost(*) in api_manager.py,  lead to the issue #4612. 

After modifying this line of code, when used Azure model, the model will return gpt-35-turbo instead none, and the update_cost(*) will look up correct data from `modelsinfo.py`.

Reply:
I have add comments for the new entries to `modelsinfo.py`, please help to review again, thanks~

Reply:
the placeholder line has been removed.

To use the deployment id for Azure ChatGPT API instead of the model name, you can call the API using the deployment id set by the user. The model name needs to be obtained through query.
Regarding auto-gpt, you can use the model name of Azure to analyze and calculate the number/cost of tokens.

Reply:
I can't figure out what you mean. Does sending the `model` along with the `deployment_id` in the API request cause the API to return the `model` as `gpt-35-turbo` in the response?

Reply:
In Auto-GPT, token calculation is based on the `model name`, in the original file, the model name was not passed to `api_manager.py `, the model `default value = ‘none’ `used by `update_cost(prompt_tokens, completion_tokens, model)`, resulting in bug#4612.

OpenAI uses the model keyword argument to specify what model to use. Azure OpenAI has the concept of [deployments] and uses the deployment_id keyword argument to describe which model deployment to use. I think the action of obtaining the model name from Azure based on the deployment ID should be performed in config.py. I did not look for the specific location.